File: ios/Classes/FlutterYidunCaptchaPlugin.m
#import "FlutterYidunCaptchaPlugin.h"

@implementation FlutterYidunCaptchaPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_yidun_captcha"
            binaryMessenger:[registrar messenger]];
  FlutterYidunCaptchaPlugin* instance = [[FlutterYidunCaptchaPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: macos/Classes/FlutterYidunCaptchaPlugin.swift
import Cocoa
import FlutterMacOS

public class FlutterYidunCaptchaPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_yidun_captcha", binaryMessenger: registrar.messenger)
    let instance = FlutterYidunCaptchaPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
